src/App/Fossa/Main.hs

```diff
@@ -87,16 +87,33 @@ appMain = do
     AnalyzeCommand AnalyzeOptions{..} -> do
       -- The branch override needs to be set here rather than above to preserve
       -- the preference for command line options.
+
+      if null analyzeBuildTargetFilters
+        then pure ()
+        else withDefaultLogger logSeverity $ logWarn "The --filter option has been deprecated. Refer to the new target exclusion feature for upgrading. --filter will be removed by v2.20.0"
+
       let analyzeOverride = override{overrideBranch = analyzeBranch <|> ((fileConfig >>= configRevision) >>= configBranch)}
-          doAnalyze destination = analyzeMain analyzeBaseDir analyzeRecordMode logSeverity destination analyzeOverride analyzeUnpackArchives analyzeJsonOutput analyzeVSIMode analyzeBuildTargetFilters
+          -- If a user enters a single target or path filtering flag, do not use any filters from the configuration file.
+          combinedFilters =
+            if null analyzeOnlyTargets && null analyzeExcludeTargets && null analyzeOnlyPaths && null analyzeExcludePaths
+              then AllFilters analyzeBuildTargetFilters includeFilters excludeFilters
+              else AllFilters analyzeBuildTargetFilters (FilterCombination (configTargetToFilter <$> analyzeOnlyTargets) analyzeOnlyPaths) (FilterCombination (configTargetToFilter <$> analyzeExcludeTargets) analyzeExcludePaths)
+            where
+              includeFilters = FilterCombination (filterTargets targetsOnly) (filterPaths pathsOnly)
+              excludeFilters = FilterCombination (filterTargets targetsExclude) (filterPaths pathsExclude)
+              filterPaths field = maybe [] field (fileConfig >>= configPaths)
+              filterTargets field = configTargetToFilter <$> maybe [] field (fileConfig >>= configTargets)
```

Comment:
[nit]

- line 100 is reaaaally long. I'd insert a newline somewhere and make fourmolu format it
- inlining the definitions of `includeFilters` and `excludeFilters` would help readability
- adding type signatures to `filterPaths` and `filterTargets` would help readability